datafusion/core/src/datasource/physical_plan/file_scan_config.rs

```diff
@@ -101,7 +101,7 @@ pub struct FileScanConfig {
     /// all records after filtering are returned.
     pub limit: Option<usize>,
     /// The partitioning columns
-    pub table_partition_cols: Vec<(String, DataType)>,
+    pub table_partition_cols: Vec<Field>,
```

Comment:
Here is the key change

Reply:
You probably want to use FieldRef not Field